de.dlr.sc.overtarget.language.ui/src/de/dlr/sc/overtarget/language/ui/handler/TargetActivationHandler.java

```diff
@@ -64,13 +78,32 @@ public Object execute(ExecutionEvent event) throws ExecutionException {
 			String outputConfig = getOutputConfigurations(project);
 			IFile targetFile;
 			String outputPath = outputConfig.replace(".", "");
+			
+			URI uri = URI.createPlatformResourceURI(file.getFullPath().toString(), true);
+			ResourceSet rs = resourceSetProvider.get(project);
+			Resource r = rs.getResource(uri, true);
+			
+			//to check if the tmodel is a target library
+			TargetFile target = (TargetFile) r.getContents().get(0);
+			ICompositeNode node = NodeModelUtils.getNode(target);
+			String targetLibraryKeyword = grammarAccess.getTargetLibraryRule().getName();
+			String nodeKeyword = node.getFirstChild().getText();
+			boolean isTargetLibrary = nodeKeyword.equals(targetLibraryKeyword);
+			
 			if (outputPath.equals("/")) {
 				targetFile = project.getFile("/" + targetName);
 			} else {
 				String targetPath = outputPath + "/" + targetName;
 				targetFile = project.getFile(targetPath);
 			}
-			if (targetFile.exists()) {
+			if (isTargetLibrary) {
```

Comment:
Maybe handle the most likely case... The one where the tmodel is a „normal“ target specification first? Otherwise it might be confusing... Sorry that’s really picky I now 😅

Reply:
Sure, that makes more sense :)